src/overlays/actors/ovl_En_Aob_01/z_en_aob_01.c

```diff
@@ -629,156 +728,205 @@ s32 func_809C2594(EnAob01* this, PlayState* play) {
     return false;
 }
 
-s32 func_809C25E4(EnAob01* this, PlayState* play) {
+/**
+ * Returns true if the race should be ended, which can happen in either of the following circumstances:
+ * - All 14 dogs have finished the race.
+ * - At least 10 dogs have finished the race, and at least 500 frames have passed since the 10th place
+ *   dog finished. This is controlled by the forceRaceEndTimer, which is set in EnAob01_Init.
+ */
+s32 EnAob01_Race_IsRaceOver(EnAob01* this, PlayState* play) {
     Actor* npc = play->actorCtx.actorLists[ACTORCAT_NPC].first;
-    s16 count = 0;
+    s16 finishedCount = 0;
 
     while (npc != NULL) {
         if ((npc->id == ACTOR_EN_RACEDOG) && (((EnRacedog*)npc)->raceStatus == RACEDOG_RACE_STATUS_FINISHED)) {
-            count++;
+            finishedCount++;
         }
         npc = npc->next;
     }
 
-    if (count >= 14) {
+    if (finishedCount >= 14) {
```

Comment:
```suggestion
    if (finishedCount >= ARRAY_COUNT(sDogInfo)) {
```
If you were to add a dog, this number should be updated right?